core/src/main/java/org/elasticsearch/tribe/TribeService.java

```diff
@@ -463,6 +487,44 @@ private ClusterState applyUpdate(ClusterState currentState, ClusterChangedEvent
                 }
             }
 
+            Set<String> changedCustomMetaDataTypeSet = tasks.stream()
+                    .map(ClusterChangedEvent::changedCustomMetaDataSet)
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toSet());
+            final List<Node> tribeNodes = TribeService.this.nodes;
+            Map<String, MetaData.Custom> mergedCustomMetaDataMap = mergeChangedCustomMetaData(changedCustomMetaDataTypeSet,
+                    customMetaDataType -> {
```

Comment:
we only get a list of `MergableCustomMetaData`s from this for all the tribeClient nodes and if we do the special casing as mentioned [here](https://github.com/elastic/elasticsearch/pull/21552#discussion_r88539383), extracting it to a static function and testing it would be difficult. I think we are better off beefing up the integration tests `TribeIT` to deal with multiple customs instead?